src/osm_configurator/model/project/calculation/attractivity_phase.py

```diff
@@ -25,4 +55,108 @@ def calculate(self, configuration_manager):
         """
-        pass
+        # Get the traffic cells
+        geojson_path: Path = configuration_manager.get_cut_out_configuration().get_cut_out_path()
+        co_parser: CutOutParser = cut_out_parser.CutOutParser()
+
+        if geojson_path is None:
+            return calculation_state_enum.CalculationState.ERROR_INVALID_CUT_OUT_DATA, "Invalid geojson file"
+
+        try:
+            cells: GeoDataFrame = co_parser.parse_cutout_file(geojson_path)
+        except IllegalCutOutException as err:
+            return calculation_state_enum.CalculationState.ERROR_INVALID_CUT_OUT_DATA, str(err.args[0])
+
+        # Delete result files
+        result_folder: Path = calculation_phase_utility.get_checkpoints_folder_path_from_phase\
+            (configuration_manager, calculation_phase_enum.CalculationPhase.ATTRACTIVITY_PHASE)
+        deleter: FileDeletion = file_deletion.FileDeletion()
+        deleter.reset_folder(result_folder)
+
+        # Iterate over all traffic cells and generate the attractivities
+        index: int
+        row: Series
+        for index, row in cells.iterrows():
+            cell_name: str = row[model_constants.CL_TRAFFIC_CELL_NAME]
+            result: calculation_state_enum
+            msg: str
+            result, msg = self._calculate_attractivity_in_traffic_cell(cell_name, configuration_manager)
+            if result != calculation_state_enum.CalculationState.RUNNING:
+                return result, msg
+
+        return calculation_state_enum.CalculationState.RUNNING, "running"
+
+    def _calculate_attractivity_in_traffic_cell(self, cell_name: str, config_manager: ConfigurationManager) \
+            -> Tuple[CalculationState, str]:
+        # get the necessary path's
+        reduction_folder: Path = calculation_phase_utility\
+            .get_checkpoints_folder_path_from_phase(config_manager,
+                                                    calculation_phase_enum.CalculationPhase.REDUCTION_PHASE)
+        attractivity_folder: Path = calculation_phase_utility \
+            .get_checkpoints_folder_path_from_phase(config_manager,
+                                                    calculation_phase_enum.CalculationPhase.ATTRACTIVITY_PHASE)
+        input_path: Path = Path(os.path.join(reduction_folder, cell_name + ".csv"))
+        output_path: Path = Path(os.path.join(attractivity_folder, cell_name + ".csv"))
+        # TODO: magic strings
+
+        # Read and create data frames
+        input_df: DataFrame = pd.read_csv(input_path)
+        output_df: DataFrame = self._init_dataframe(config_manager.get_category_manager())
+
+        # Iterate over all elements in this cell
+        index: int
+        element: Series
+        for index, element in input_df.iterrows():
+            try:
+                output_df = self._calculate_attractivity_for_element(element, output_df,
```

Comment:
This is bad style you shouldn't do this(this leads to evry poor performance), look below at method implementation for details.